### ISSUE TITLE
The total loss doesn't equal to the sum of the four terms.

### ISSUE THREAD

**Issue by catsdogone** [rbgirshick/py-faster-rcnn#70]
The total loss doesn't equal to the sum of the four terms, especially " Iteration 48460, loss = 5.59026", it is weird:
...
I0125 09:26:43.294986 5561 solver.cpp:571] Iteration 48420, lr = 5e-05
I0125 09:27:17.914181 5561 solver.cpp:242] Iteration 48440, loss = 0.891995
I0125 09:27:17.914214 5561 solver.cpp:258] Train net output #0: loss_bbox = 0.204296 (* 1 = 0.204296 loss)
I0125 09:27:17.914221 5561 solver.cpp:258] Train net output #1: loss_cls = 0.177461 (* 1 = 0.177461 loss)
I0125 09:27:17.914225 5561 solver.cpp:258] Train net output #2: rpn_cls_loss = 0.0711229 (* 1 = 0.0711229 loss)
I0125 09:27:17.914229 5561 solver.cpp:258] Train net output #3: rpn_loss_bbox = 0.126131 (* 1 = 0.126131 loss)
I0125 09:27:17.914234 5561 solver.cpp:571] Iteration 48440, lr = 5e-05
I0125 09:27:49.331830 5561 solver.cpp:242] Iteration 48460, loss = 5.59026
I0125 09:27:49.331864 5561 solver.cpp:258] Train net output #0: loss_bbox = 0.0632553 (* 1 = 0.0632553 loss)
I0125 09:27:49.331871 5561 solver.cpp:258] Train net output #1: loss_cls = 0.0835277 (* 1 = 0.0835277 loss)
I0125 09:27:49.331876 5561 solver.cpp:258] Train net output #2: rpn_cls_loss = 0.00019457 (* 1 = 0.00019457 loss)
I0125 09:27:49.331881 5561 solver.cpp:258] Train net output #3: rpn_loss_bbox = 0.00126853 (* 1 = 0.00126853 loss)
I0125 09:27:49.331886 5561 solver.cpp:571] Iteration 48460, lr = 5e-05

**Comment by happyharrycn** [comment]
The displayed loss is averaged across multiple iterations (e.g. 100). This is set by average_loss (100) in solver.prototxt.

**Comment by catsdogone** [comment]
But they still should be equal even the average, isn't it?

**Comment by imbinwang** [comment]
@catsdogone Could you show me your loss curves? I got invalid loss curves of rpn related losses, the training losses of rpn don't converge on my dataset. I want to know what a valid loss curves look like.

Thank you.

**Comment by Sancerio** [comment]
any idea on this, @happyharrycn ? I also get this issue

**Comment by happyharrycn** [comment]
The train net output (losses at different branch) is the network output at current iteration. The loss is an running average of the sum of all losses over multiple iterations. And that is why the sum of net output is not equal to the loss.

**Comment by catsdogone** [comment]
@happyharrycn  But when I set the iter_size: 1 instead of iter_size : 2 in the solver.prototxt, the loss equals to the sum of the four terms
